### DISCUSSION TITLE
How can I put together targetRevisions of multiple Application?

### DISCUSSION THREAD

**Discussion by lirlia** [argoproj/argo-cd#7269] [Q&A]
I'm using ArgoCD in an App-of-Apps pattern.

I have many application manifests  like this
- argocd.yaml
- argocd-config.yaml
- prometheus.yaml
- ...

Those manifest has **same targetRevision** , for example release-20210921 git tag.
I think that It's lots of waste, so I'd like to define **targetRevision** only 1 time.

How can I do that by ArgoCD ( or Kustomize) ?

**Comment by wanghong230** [comment]
Do you want to try to move those files into a base?

**Reply by lirlia** [reply]
Thanks your reply.

I already use base, but I want to change tag any time on release. 
So I need to change **targetRevision** per update.

**Reply by wanghong230** [reply]
1. You may consider using helm as the helm is better at doing the substitution via values.yaml. You can provide targetRevision in the values.yaml, and reference it in all application YAMLs.

2. You can explore the usage of ApplicationSet which might be perfect answer to you. I assume you are deploying multiple applications from the same repo with the same branch/revision.
https://argocd-applicationset.readthedocs.io/en/stable/Generators-Git/
https://argocd-applicationset.readthedocs.io/en/stable/

**Reply by lirlia** [reply]
Thanks!
I'll use ApplicationSet !
